### ISSUE TITLE
Filtered PullReplication on a channel

### ISSUE THREAD

**Issue by NitzDKoder** [couchbase/couchbase-lite-android#590]
Hi Folks,

I have android LITE with below pull setup.

Replication pullReplication = database.createPullReplication(syncUrl);
pullReplication.setContinuous(true);
Authenticator auth = new BasicAuthenticator(user, "password");
pullReplication.setAuthenticator(auth); 
userChannel = Arrays.asList("user1");
pullReplication.setChannels(userChannel); 
pullReplication.setFilter("designdoc/itemFilter");
Map params = new HashMap();
params.put("searchid", "item");
pullReplication.setFilterParams(params);
pullReplication.start();
pullReplication.addChangeListener(this);

How do i deal with filters if i am using couchbase server.?

Channel and filter can work together?

Basically I want to pull some filtered docs which belongs to a channel.

Have created views and done some emit http://docs.couchbase.com/admin/admin/UI/ui-views-editor.html.

Is the the way the filters work if i am not using couchDB?if so how to set filters in lite.

Something similar to https://github.com/couchbase/couchbase-lite-android/issues/200

Little puzzled help me out.

Thanks
Ntihin

**Comment by hideki** [comment]
Discussed in Couchbase Forum: https://forums.couchbase.com/t/filtered-pullreplication-on-a-channel/3601

> pullReplication.setFilter("designdoc/itemFilter");

It seem that the code tries to use Couchbase Server's filter. CBL Android's Replicator filter is independent from Couchbase Server filter. This is not supported.

**Comment by NitzDKoder** [comment]
@hideki Now pull docs based on docid is supportted now? is there UT done for the same? pls confirm..

https://github.com/couchbase/couchbase-lite-java-core/blob/release/1.2.0/src/main/java/com/couchbase/lite/replicator/PullerInternal.java

changeTracker.setDocIDs(documentIDs);

**Comment by hideki** [comment]
@NitzDKoder 
Replicator supports docIDs by `public void Replication.setDocIds(List<String> docIds)`.
https://github.com/couchbase/couchbase-lite-java-core/blob/262a8fab84a1ca239cb3b9735c3fed7209c28909/src/main/java/com/couchbase/lite/replicator/Replication.java#L680

**Comment by NitzDKoder** [comment]
@hideki 
		```
pullReplication.setFilter("_doc_ids");
		pullReplication.setDocIds(Arrays.asList(docIds));
		pullReplication.start();
```

This should pull a doc based on docid? CBL knows the rev? or decides on rev and then fetches?

**Comment by hideki** [comment]
@NitzDKoder 

> This should pull a doc based on docid? 

YES

> CBL knows the rev? or decides on rev and then fetches?

push and pull all revisions

`pullReplication.setFilter("_doc_ids");` <- does not need to set
